### ISSUE TITLE
using declaration example containing goto doesn't compile

### ISSUE THREAD

**Issue by RexJaeschke** [dotnet/docs#33669]
I am integrating this proposal into the Ecma C# standard.

This proposal contains the following text, along with an example:

> There are no restrictions around goto, or any other control flow construct in the face of a using declaration. Instead the code acts just as it would for the equivalent using statement: <example>..

The example shows `goto` jumping to label `target`. However, this results in the following error:

> CS8649: A goto cannot jump to a location before a using declaration within the same block.

1. Does this mean that the paragraph cited above (and its example) should be removed? 
2. If not, how should it (and the example) be changed?
3. What about the "other control flow constructs;" is there anything we can still say about them?

Rex

---
#### Document Details

⚠ *Do not edit this section. It is required for learn.microsoft.com ➟ GitHub issue linking.*

* ID: bc900853-b666-5815-9d69-e408af3e3900
* Version Independent ID: 77f3af83-d0ca-7d48-d06d-0f28f5f05a49
* Content: [Pattern based using and using declarations - C# 8.0 draft specifications](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-8.0/using)
* Content Source: [proposals/csharp-8.0/using.md](https://github.com/dotnet/csharplang/blob/main/proposals/csharp-8.0/using.md)
* Product: **dotnet-csharp**
* Technology: **csharp-spec**
* GitHub Login: @dotnet-bot
* Microsoft Alias: **wiwagn**

**Comment by BillWagner** [comment]
ping @jcouv @333fred 

I tried the sample in question:

```csharp
{
    using var f1 = new FileStream("...");
  target:
    using var f2 = new FileStream("...");
    if (someCondition) 
    {
        // Causes f2 to be disposed but has no effect on f1
        goto target;
    }
}
```

it does generate a compiler error.

The equivalent using statement doesn't generate an error:

```csharp
{
    using (var f1 = new FileStream("..."))
    {
  target:
        using )var f2 = new FileStream("..."))
        {
            if (someCondition) 
            {
                // Causes f2 to be disposed but has no effect on f1
                goto target;
            }
        }
    }
}
```

Who would have the correct details on how we should update the spec?

**Comment by 333fred** [comment]
@chsienki should.

**Comment by jcouv** [comment]
Yup, looks like the [speclet](https://github.com/dotnet/csharplang/blob/main/proposals/csharp-8.0/using.md) is out-of-date. There are definitely restrictions around jumps.

**Comment by chsienki** [comment]
Yes, it was decided not to allow jumping 'back out' of a logical scope either, even though you can with a using statement. https://github.com/dotnet/csharplang/blob/3c8559f186d4c5df5d1299b0eaa4e139ae130ab6/meetings/2018/LDM-2018-12-03.md#goto-and-using-declarations

**Comment by BillWagner** [comment]
Closing as the C# 8 behavior is being standardized this month.
